bots/po-refresh

```diff
@@ -56,11 +57,67 @@ def run(context, verbose=False, **kwargs):
     work = os.path.abspath(line.strip())
 
     # Do the various updates
-    cmd = [ "make", "upload-pot", "download-po" ]
+    cmd = [ "make", "upload-pot", "clean-po", "download-po" ]
     if verbose:
         sys.stderr.write("+ " + " ".join(cmd) + "\n")
     subprocess.check_call(cmd, cwd=work)
 
+    local_branch = None
+    user = None
+    current_manifest = None
+    current_linguas = []
+    language_map = {}
+
+    # Get locale from language-map file. Fail with keyError when not found
+    def get_locale(file_name):
+        language = file_name.split("/")[1][:-3]
+        return language_map[language]
+
+    def add_and_commit_lang(name, language, action):
+        with open("po/LINGUAS", "w", encoding='utf-8') as lngs:
+            print("\n".join(current_linguas), file=lngs)
+        with open("pkg/shell/manifest.json.in", "w", encoding='utf-8') as mnfst:
+            text = json.dumps(current_manifest, ensure_ascii=False, indent=4)
+            print(text, file=mnfst)
+        execute("git", "add", "--", "po/LINGUAS", "pkg/shell/manifest.json.in", name)
+        return task.branch(context, "po: {0} '{1}' language".format(action, language),
+                           pathspec=None, branch=local_branch, push=False, **kwargs)
+
+    # Build language map
+    with open("po/language_map.txt", "r") as lm:
+        for line in lm:
+            items = line.strip().split(":")
+            language_map[items[0]] = items
```

Comment:
These files tend to get trailing newlines, and thus you'll get an `IndexError` there. So perhaps `continue` after `line.strip()` if it's falsy.